.github/workflows/continuous-integration.yml

```diff
@@ -22,7 +22,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
-        run: python -m pip install .[dev]
+        run: python -m pip install -r requirements.dev.txt && python -m pip install .[dev] --no-deps
```

Comment:
Ideally this would be paired with something like Dependabot to refresh these dependencies on a schedule.